infrastructure/smart-contracts/contracts/upgrades/node-license/NodeLicense8.sol

```diff
@@ -883,8 +887,9 @@ contract NodeLicense8 is ERC721EnumerableUpgradeable, AccessControlUpgradeable
         usedTransferIds[transferId] = true;
         
         for(uint256 i; i < tokenIdsLength; i++){
-            require(Referee10(refereeAddress).assignedKeyToPool(tokenIds[i]) == address(0), "Cannot transfer staked key");
+            if(Referee10(refereeAddress).assignedKeyToPool(tokenIds[i]) != address(0)) revert CannotTransferStakedKey();
             super.safeTransferFrom(msg.sender, to, tokenIds[i]);
+            Referee10(refereeAddress).updateBulkSubmissionOnTransfer(msg.sender, to, tokenIds[i]);
```

Comment:
This must not happen in the loop for each key.

Reply:
Moved to after the loop after all transfers have been processed.